internal/tsoptions/parsinghelpers.go

```diff
@@ -507,9 +511,59 @@ func mergeCompilerOptions(targetOptions, sourceOptions *core.CompilerOptions) *c
 			targetField.Set(sourceField)
 		}
 	}
+
+	// Then handle explicitly set null values using the raw JSON (if provided)
+	if rawSource != nil {
+		if rawMap, ok := rawSource.(*collections.OrderedMap[string, any]); ok {
+			if compilerOptionsRaw, exists := rawMap.Get("compilerOptions"); exists {
+				if compilerOptionsMap, ok := compilerOptionsRaw.(*collections.OrderedMap[string, any]); ok {
+					handleExplicitNullValues(targetOptions, compilerOptionsMap)
+				}
+			}
+		}
+	}
```

Comment:
This way of doing things is inefficient; what we should do instead is before the "normal merge", collect a set of all explicitly `null` entries from the raw options, then in the normal merge, check to see if that field is null'd out (extracting its name via the struct tag as done in handleExplicitNullValues) and then explicitly zero it instead of copying the value.